File: Exercise/Medium/15. Build Toy Cars/Code/Build Toy Cars.py
import sys

def main():
    tires, bodies, figures = map(int, sys.stdin.readline().split())

    cars_by_tires = tires // 4
    cars_by_bodies = bodies
    cars_by_figures = figures // 2

    complete_cars = min(cars_by_tires, cars_by_bodies, cars_by_figures)

    print(complete_cars)

if __name__ == "__main__":
    main()

File: Exercise/Medium/15. Build Toy Cars/Code/Build Toy Cars.c
#include <stdio.h>

int main() {
    int tires, bodies, figures;
    
    scanf("%d, %d, %d", &tires, &bodies, &figures);

    int cars_by_tires = tires / 4;
    int cars_by_bodies = bodies;
    int cars_by_figures = figures / 2;
    int complete_cars = cars_by_tires;
    if (cars_by_bodies < complete_cars) {
        complete_cars = cars_by_bodies;
    }
    if (cars_by_figures < complete_cars) {
        complete_cars = cars_by_figures;
    }

    printf("%d\n", complete_cars);

    return 0;
}

